frontend/model/contracts/group.js

```diff
@@ -141,6 +143,63 @@ function isActionYoungerThanUser (actionMeta: Object, userProfile: ?Object): boo
   return compareISOTimestamps(actionMeta.createdDate, userProfile.joinedDate) > 0
 }
 
+function updateGroupStreaks ({ state, getters }) {
+  const streaks = state.streaks
+  const cPeriod = getters.groupSettings.distributionDate
+  const thisPeriodPayments = getters.groupPeriodPayments[cPeriod]
+
+  // if there's no period payments set yet, no need to update streak values.
+  if (!thisPeriodPayments) return
+
+  const thisPeriodDistribution = thisPeriodPayments.lastAdjustedDistribution || adjustedDistribution({
+    distribution: unadjustedDistribution({
+      haveNeeds: getters.haveNeedsForThisPeriod(cPeriod),
+      minimize: getters.groupSettings.minimizeDistribution
+    }) || [],
+    payments: getters.paymentsForPeriod(cPeriod),
+    dueOn: getters.dueDateForPeriod(cPeriod)
+  })
```

Comment:
One of the issues with DRY violations is, this code must be kept in sync with the other copy of it in `updateAdjustedDistribution`. For example, here we're missing a `.filter` to filter for active members only.

I'm still thinking on how to fix the DRY violation, but in case this helps you figure it out before me, I'm leaving this comment here now.

Reply:
OK, yeah, I can't figure out a way to cleanly DRY this. However, that `.filter` should be copied over here so that it matches the other code.

Also, could you please a comment? something along the lines of:

```js
// TODO: see if it's possible to DRY this with the code inside of updateAdjustedDistribution
```

Reply:
done it!